### PR TITLE
Cone Uprighter

### PR DESCRIPTION
**Pull request by bireyani**
Created skeleton for the cone uprighter code
created new constants in the constants file

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ConeUprighter.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class ConeUprighter extends SubsystemBase {
+    /** Creates a new ConeUprighter. */
+    private CANSparkMax ConeUprightMotor;
```
Comment (PNovak619) [inline src/main/java/frc/robot/subsystems/ConeUprighter.java]: Modify Coneuprighter motor variable naming convention

File: src/main/java/frc/robot/subsystems/ConeUprighter.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
```
Comment (PNovak619) [inline src/main/java/frc/robot/subsystems/ConeUprighter.java]: please remove comment block or add  something useful here.

File: src/main/java/frc/robot/subsystems/ConeUprighter.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+     * 
+     */
+    public ConeUprighter() {
+        ConeUprightMotor = new CANSparkMax(Constants.CONE_UPRIGHTER_CANSPARKMAX, Constants.UPRIGHT_MOTOR_TYPE);
+        // TODO: Figure out motor type, code movement of motor, code buttons
+    }
+
+    public void operatorController() {
```
Comment (PNovak619) [inline src/main/java/frc/robot/subsystems/ConeUprighter.java]: please remove this method

File: src/main/java/frc/robot/subsystems/ConeUprighter.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    }
+
+    public void rotateFull() {
```
Comment (PNovak619) [inline src/main/java/frc/robot/subsystems/ConeUprighter.java]: Commands are coded in commands/*.java

File: src/main/java/frc/robot/subsystems/ConeUprighter.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    private CANSparkMax ConeUprightMotor;
+
+    /**
+     * 
+     */
+    public ConeUprighter() {
+        ConeUprightMotor = new CANSparkMax(Constants.CONE_UPRIGHTER_CANSPARKMAX, Constants.UPRIGHT_MOTOR_TYPE);
+        // TODO: Figure out motor type, code movement of motor, code buttons
+    }
+
```
Comment (PNovak619) [inline src/main/java/frc/robot/subsystems/ConeUprighter.java]: create method to control motor speeds

**Review by PNovak619 — CHANGES_REQUESTED** [review]
please see inline comments.  Thank you!

File: src/main/java/frc/robot/subsystems/ConeUprighter.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+public class ConeUprighter extends SubsystemBase {
+    /** Creates a new ConeUprighter. */
+    private CANSparkMax coneUprightMotor;
+
+    public ConeUprighter() {
+        coneUprightMotor = new CANSparkMax(Constants.CONE_UPRIGHTER_CANSPARKMAX, Constants.UPRIGHT_MOTOR_TYPE);
+        // TODO: Figure out motor type, code movement of motor, code buttons
+    }
+    public void UprightSpeed() {
```
Comment (PNovak619) [inline src/main/java/frc/robot/subsystems/ConeUprighter.java]: modify to set motor speed.
change to camel case formatting

File: src/main/java/frc/robot/subsystems/ConeUprighter.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class ConeUprighter extends SubsystemBase {
+    /** Creates a new ConeUprighter. */
+    private CANSparkMax coneUprightMotor;
+
```
Comment (PNovak619) [inline src/main/java/frc/robot/subsystems/ConeUprighter.java:15]: add TODO to add encoder once we know motor controller type

**Review by PNovak619 — CHANGES_REQUESTED** [review]
please see inline comments.  Thank you!

**Review by PNovak619 — APPROVED** [review]

**Review by programmer94 — APPROVED** [review]
Great work! Thanks for your patience.

**Pull request merged**